debezium-connector-postgres/src/main/java/io/debezium/connector/postgresql/RecordsSnapshotProducer.java

```diff
@@ -207,26 +208,30 @@ private void takeSnapshot(BlockingConsumer<ChangeEvent> consumer) {
 
             // and mark the start of the snapshot
             sourceInfo.startSnapshot();
-            sourceInfo.update(xlogStart, clock().currentTimeInMicros(), txId, null);
+            // use the old xmin, as we don't want to update it if in xmin recovery
+            sourceInfo.update(xlogStart, clock().currentTimeInMicros(), txId, null, sourceInfo.xmin());
 
             logger.info("Step 3: reading and exporting the contents of each table");
             AtomicInteger rowsCounter = new AtomicInteger(0);
-            final Map<TableId, String> selectOverrides = getSnapshotSelectOverridesByTable();
 
             for(TableId tableId : schema.tableIds()) {
                 long exportStart = clock().currentTimeInMillis();
                 logger.info("\t exporting data from table '{}'", tableId);
                 try {
-                    // DBZ-298 Quoting name in case it has been quoted originally; it doesn't do harm if it hasn't been quoted
-                    final String selectStatement = selectOverrides.getOrDefault(tableId, "SELECT * FROM " + tableId.toDoubleQuotedString());
-                    logger.info("For table '{}' using select statement: '{}'", tableId, selectStatement);
-
-                    connection.queryWithBlockingConsumer(selectStatement,
-                            this::readTableStatement,
-                            rs -> readTable(tableId, rs, consumer, rowsCounter));
-                    logger.info("\t finished exporting '{}' records for '{}'; total duration '{}'", rowsCounter.get(),
-                            tableId, Strings.duration(clock().currentTimeInMillis() - exportStart));
-                    rowsCounter.set(0);
+                    final String selectStatement = snapshotter.buildSnapshotQuery(tableId);
+                    if (selectStatement == null || selectStatement.isEmpty()) {
```

Comment:
If we support omitting of query for table I thing we should use `Optional` insted of `null`

Reply:
yeah, I wasn't totally sure on this... it seemed like a sane thing to do, just for a defensive programming for a user controlled class, but changing the signature communicates that more if that feels more like "feature" rather than an implementation detail that prevents a NPE.